### ISSUE TITLE
Spans are forcibly ended after transaction ends

### ISSUE THREAD

**Issue by dgieselaar** [elastic/apm-agent-rum-js#1023]
I'm seeing `span.type:external.truncated.truncated` in my local APM data. From what I gather on Slack and github, the reason for that is that spans are ended by the agent if the transaction ends. It looks like that is no longer needed, per this Slack message of @axw:

> Older agents did not allow spans to outlive their transaction. When a transaction was ended, any ongoing spans were forcibly ended then, and ".truncated" appended to their type to give some indication of problematic instrumentation.
> We moved to a newer protocol which decoupled the lifetime of spans from their transactions, so we don't do that any more.

Not sure if this choice is intentional for the RUM agent, but the code is two years old, so maybe an oversight.

**Comment by vigneshshanmugam** [comment]
At least for the RUM agent, we would still need to truncate the spans and include them as part of the transaction to make sure the compression and other decoding logic in server works as expected. Also, RUM had never sent spans as a separate event from the start ,so not sure if it makes sense to diverge now unless there is a huge advantage in doing so. 

More details about compression - #768 

@jalvz  Please correct me if i am wrong. 

We are working on changing the details around when transaction would end since the introduction of session, so things would change in future and we would see less of the truncated spans.

**Comment by jalvz** [comment]
Agreed @vigneshshanmugam , IIUC Andrew is talking about the streaming protocol that Rum does not use
